src/game/items/CItem.cpp

```diff
@@ -2271,14 +2271,14 @@ void CItem::r_WriteMore1(CSString & sVal)
         case IT_GRASS:
         case IT_ROCK:
         case IT_WATER:
-            sVal = g_Cfg.ResourceGetName(m_itResource.m_ridRes);
+            sVal = g_Cfg.ResourceGetName(CResourceID(RES_REGIONRESOURCE, m_itResource.m_ridRes.GetResIndex())); //Changed to fix issue but it is not implemented.
```

Comment:
The problem is as much as we put CResourceID& rid in input of function, it will throw an exception, also as it's more value, it may be impossible to get ItemDef from rid (I am not sure).

I can add another ResourceGetName() function under CItem.h to check if item or itembase has CAN_I_SCRIPTED can flag if we really need to add CAN_I_SCRIPTED.

For a static function, I can create a new function with CResourceIDBase input to avoid errors and do checks.

Reply:
As long as you pass references and accessing raw internal data i think you shouldn't get errors

Reply:
I tried it but as soon as I put any CResourceBase input to CResourceID&, it throw an error that rid should be valid resource. It's because of Copy Constructor of CResourceBase.

Reply:
Okay then i'll address that in another commit i guess